src/approaches/interactive_learning_approach.py

```
@@ -67,41 +69,165 @@ def learn_from_offline_dataset(self, dataset: Dataset) -> None:
+        # try other strategies. But one nice thing about random selection
+        # is that we're not making a hard commitment to the agent having
+        # control over which train task it gets to use.
+        return self._rng.choice(len(self._train_tasks))
+
+    def _create_interaction_action_strategy(
+        self, train_task_idx: int
+    ) -> Tuple[Callable[[State], Action], Callable[[State], bool]]:
+        """Returns an action policy and a termination function."""
+        if CFG.interactive_action_strategy == "glib":
+            return self._create_glib_interaction_strategy(train_task_idx)
+        raise NotImplementedError("Unrecognized interactive_action_strategy:"
+                                  f" {CFG.interactive_action_strategy}")
+
+    def _create_query_policy(
+            self, train_task_idx: int) -> Callable[[State], Optional[Query]]:
+        """Returns a query policy."""
+        del train_task_idx  # unused right now, but future policies may use
+        if CFG.interactive_query_policy == "strict_best_seen":
+            return self._create_best_seen_query_policy()
+        raise NotImplementedError("Unrecognized interactive_query_policy:"
+                                  f" {CFG.interactive_query_policy}")
+
+    def _create_glib_interaction_strategy(
+        self, train_task_idx: int
+    ) -> Tuple[Callable[[State], Action], Callable[[State], bool]]:
+        """Find the most interesting reachable ground goal and plan to it."""
         init = self._train_tasks[train_task_idx].init
         # Detect and filter out static predicates.
         static_preds = utils.get_static_preds(self._nsrts,
                                               self._predicates_to_learn)
         preds = self._predicates_to_learn - static_preds
-        # Find acting policy for the request.
-        task_list = glib_sample(init, preds, self._dataset)
-        assert task_list
+        # Sample possible goals to plan toward.
```

You may want to make this fall back to random by making this sample_subsets function return a generator, and then `find_first_solvable` loops over that generator and if none of the goals are reachable, fall back. Can be done in  a different PR too.

---

I'll make it a generator, but we need to sort these, so it won't make a difference here.